### PR TITLE
Fix fortran library compilation

### PR DESCRIPTION
**Pull request by clementval**
With @pallaskat, we discover that the source of my problem with serialbox from fortran were a wrong compilation of the Fortran library. 

I moved the `include_directories`statement were they are really needed. Like this, they are not passed to the Fortran compiler during the compilation of the library. 

I have a small Fortran example that I can put on the repo with its own CMake. Should I put it @cosunae @andyspiros @pspoerri ? Is so, I'll do a PR with it after this one is merged (hopefully)

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1,9 @@
+CMakeCache.txt
+CMakeFiles
+CMakeScripts
+Makefile
+cmake_install.cmake
+install_manifest.txt
+*.a
+*.mod
+*.so
```
Comment (andyspiros) [inline .gitignore:9]: We can probably put `build/` in the .gitignore and remove the CMake files (all the non-starred files above).

Comment (clementval) [inline .gitignore:9]: But those file can be generated at the root of the repo ...

**Comment by andyspiros** [comment]
It is fine to me, but I don't understand why it is not working if you put the `include_directories` in the computation of the fortran part too.

**Comment by cosunae** [comment]
sounds good to me. Just for my understanding, why the include directories is spoiling the fortran compilation?
I think adding your fortran example would be a good starting point to have a set of meaningful example for testing, showing usage to users.

**Comment by clementval** [comment]
@andyspiros @cosunae I don't have the exact answer for that but anyway with Christophe we found out that this was the reason why the fortran library was giving me so much segfault.
Anyway, it is better to not pass includes option were they are not needed.

**Comment by cosunae** [comment]
good thanks

**Comment by andyspiros** [comment]
It looks good, you can merge from my point of view.

**Pull request merged**